docs/standard/serialization/binaryformatter-migration-guide/read-nrbf-payloads.md

```diff
@@ -0,0 +1,237 @@
+---
+title: "BinaryFormatter migration guide: Read BinaryFormatter (NRBF) payloads"
+description: "Safely read payloads created by BinaryFormatter in '.NET Remoting: Binary Format (NRBF).'"
+ms.date: 5/31/2024
+no-loc: [BinaryFormatter, Serialization]
+dev_langs:
+  - CSharp
+helpviewer_keywords:
+  - "BinaryFormatter"
+  - "serializing objects"
+  - "serialization"
+  - "objects, serializing"
+---
+
+# Read BinaryFormatter (NRBF) payloads
+
+BinaryFormatter used the [.NET Remoting: Binary Format](/openspecs/windows_protocols/ms-nrbf/) for serialization. This format is known by its abbreviation of MS-NRBF or just NRBF. A common challenge involved in migrating from BinaryFormatter is dealing with BinaryFormatter payloads persisted to storage as reading these payloads previously required BinaryFormatter. Some systems need to retain the ability to read these payloads for gradual migrations to new serializers while avoiding a reference to BinaryFormatter itself.
+
+As part of .NET 9, a new `NrbfDecoder` class was introduced to decode NRBF payloads without performing _deserialization_ of the payload. This API can safely be used to decode trusted or untrusted payloads without any of the risks that BinaryFormatter deserialization carries. However, `NrbfDecoder` merely decodes the data into structures an application can further process. Care must be taken when using `NrbfDecoder` to safely load the data into the appropriate instances.
+
+## NrbfDecoder
+
+`NrbfDecoder` is part of the new [System.Formats.Nrbf](https://www.nuget.org/packages/System.Formats.Nrbf) NuGet package. It targets not only .NET 9, but also older monikers like .NET Standard 2.0 and .NET Framework. That multi-targeting makes it possible for everyone who uses a supported version of .NET to migrate away from `BinaryFormatter`.
+
+`NrbfDecoder` can read payloads that were serialized with `BinaryFormatter` using `FormatterTypeStyle.TypesAlways` (the default). `NrbfDecoder` cannot be used for the output of BinaryFormatter for any other `FormatterTypeStyle`.
```

Comment:
This is not entirely true.  We support `FormatterTypeStyle.TypesAlways` and `FormatterTypeStyle.TypesAlways | FormatterTypeStyle.XsdString`

https://github.com/dotnet/runtime/blob/483db3e224dc2a942ca3805ffbd82af13d230236/src/libraries/System.Formats.Nrbf/tests/ReadTests.cs#L40-L44

Reply:
Suggestions on how to reword it?

Reply:
I think we can just drop the '... cannot be used ... for any other `FormatterTypeStyle`' sentence. I'll delete it.